plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/AddToIndexHandler.java

```diff
@@ -0,0 +1,318 @@
+            }
+        };
+        gitStatus(callback);
+    }
+
+    private void gitStatus(AsyncRequestCallback<Status> callback) {
+        gitService.status(appContext.getWorkspaceId(),
+                          appContext.getCurrentProject().getRootProject(),
+                          callback);
+    }
+
+    private void addToIndex(List<String> filesToAdd, final Notification successNotification) {
+        final ProjectConfigDto project = appContext.getCurrentProject().getRootProject();
+        try {
+            gitService.add(appContext.getWorkspaceId(), project, false, filesToAdd,
+                           new RequestCallback<Void>() {
+                               @Override
+                               protected void onSuccess(Void result) {
+                                   notificationManager.notify(successNotification);
+                               }
+
+                               @Override
+                               protected void onFailure(Throwable exception) {
+                                   notificationManager.notify(locale.addFailed(), FAIL, true, project);
+                               }
+                           });
+        } catch (WebSocketException exception) {
+            notificationManager.notify(locale.addFailed(), FAIL, true, project);
+        }
+    }
+
+    private String normalizePath(String path) {
+        String projectPath = appContext.getCurrentProject().getRootProject().getPath();
+
+        String pattern = path;
+        if (path.startsWith(projectPath)) {
+            pattern = pattern.replaceFirst(projectPath, "");
+        }
+        if (pattern.startsWith("/")) {
+            pattern = pattern.replaceFirst("/", "");
+        }
+        return pattern;
+    }
+
+    private boolean wasRenamed(List<String> missing, String oldName) {
+        for (String renaming : missing) {
+            if (renaming.contains(oldName)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String getNodePathName(ResourceBasedNode node) {
+        if (node instanceof ModuleNode) {
+            return  ((ModuleNode)node).getStorablePath();
+        } else if (node instanceof FolderReferenceNode) {
+            return  ((FolderReferenceNode)node).getStorablePath();
+        } else if (node instanceof FileReferenceNode) {
+            return  ((FileReferenceNode)node).getStorablePath();
+        } else {
+            return "";
+        }
+    }
+
+    private boolean isUnderGit() {
+        //TODO rework checking for git mixin when IDEX-3725 will be fixed
+        final ProjectConfigDto project = appContext.getCurrentProject().getRootProject();
+        Map<String, List<String>> attributes = project.getAttributes();
+        return attributes.containsKey("vcs.provider.name") && attributes.get("vcs.provider.name").contains("git");
```

Comment:
It would be better to move [these](https://github.com/codenvy/che-core/blob/project_api_4/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectType.java#L24-L26) constants to the shared class (see example [here](https://github.com/codenvy/che-core/blob/project_api_4/platform-api/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/shared/Constants.java)) and use it there.